### PR TITLE
Adding generics for all classes

### LINKED ISSUE #12: Add generic metadata support for classes

### PR DESCRIPTION
**Pull request by nathansit**
looking for review LFR
if this is correct I will add it to the other classes
Eventually Closes #12

### CODE REVIEW

File: src/Book.ts
Comment (gideontong) [inline src/Book.ts:1]: Unfortunately, this isn't correct at all. I don't need a generic class type, as we don't have types of Books (as, Books are not type of Old Testament or New Testament, rather, those are classifications of Book), instead, we need a metadata field (and thus, you need to rename the PR), that allows generic data to be added. This should probably be of just type Object.

**Review by gideontong — CHANGES_REQUESTED** [review]

File: src/classes/Passage.ts [hunk 1]
```diff
@@ -1,4 +1,12 @@
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
+<<<<<<< HEAD:src/Book.ts
```
Comment (gideontong) [inline src/classes/Passage.ts]: You left a merge conflict in the PR

**Review by gideontong — CHANGES_REQUESTED** [review]

File: src/classes/Passage.ts (hunk 1 above)
Comment (nathansit) [reply src/classes/Passage.ts]: Ahh good catch idk why but I thought I could leave it in

File: src/classes/Passage.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-class Passage {
+class Book<T> {
```
Comment (gideontong) [inline src/classes/Passage.ts]: This is too generic

**Review by gideontong — CHANGES_REQUESTED** [review]

**Review by gideontong — APPROVED** [review]

**Pull request merged**